lib/charms/layer/basic.py

```diff
@@ -290,7 +292,10 @@ def _load_installed_versions(pip):
 def _load_wheelhouse_versions():
     versions = {}
     for wheel in glob('wheelhouse/*'):
-        pkg, ver = os.path.basename(wheel).rsplit('-', 1)
+        if wheel.endswith('.whl'):
+            pkg, ver, _ = os.path.basename(wheel).split('-', 2)
```

Comment:
For reference: assuming the file name convention from PEP 427 is used here

https://peps.python.org/pep-0427/#file-name-convention

```
os.path.basename('pip-22.2.2-py3-none-any.whl').split('-', 2)
['pip', '22.2.2', 'py3-none-any.whl']
```

Reply:
Yeah, I'll add a comment to document that, thank you for the pointer!